ainstein_radar_drivers/src/radar_interface_o79_can.cpp

```diff
@@ -124,12 +125,21 @@ namespace ainstein_radar_drivers
             ROS_DEBUG( "received message with unknown id: %02x", msg.id );
           }
 
-        if(targets_received >= targets_to_come)
+        if( targets_received == targets_to_come )
           {
-            pub_radar_data_raw_.publish( radar_data_msg_ptr_raw_ );
-	          pub_radar_data_tracked_.publish( radar_data_msg_ptr_tracked_ );
+            if( ( target_type == 0 ) || ( ( ros::Time::now() - radar_data_msg_ptr_raw_->header.stamp ) > t_raw_timeout ) )
+              {
+                // if we get a tracked frame, and it's been a while since we got a
+                // raw frame, publish an empty raw frame to clear the display
+                pub_radar_data_raw_.publish( radar_data_msg_ptr_raw_ );
+              }
+            else if( target_type == 1 )
+              {
+                pub_radar_data_tracked_.publish( radar_data_msg_ptr_tracked_ );
+              }
```

Comment:
As discussed, this was done because the tracked messages are sent from the radar periodically while the raw are not, so triggering a raw ROS message every time tracked data is available from the radar makes both raw and tracked periodic. We should make the radar send raw data periodically in the future and remove this sort of workaround, but it works for now.